src/algokit/cli/tasks/analyze.py

```diff
@@ -0,0 +1,230 @@
+import json
+import logging
+import re
+from pathlib import Path
+
+import click
+
+from algokit.core.tasks.analyze import (
+    TEALER_SNAPSHOTS_ROOT,
+    generate_report_filename,
+    generate_summaries,
+    generate_tealer_command,
+    has_baseline_diff,
+    load_tealer_report,
+    prepare_artifacts_folders,
+    run_tealer,
+)
+from algokit.core.utils import run_with_animation
+
+logger = logging.getLogger(__name__)
+
+
+def display_analysis_summary(analysis_results: dict) -> None:
+    """
+    Display the summary of the analysis results.
+
+    Args:
+        analysis_results (dict): Dictionary containing analysis results.
+    """
+    impact_frequency: dict = {}
+    for file_path, result_rows in analysis_results.items():
+        click.echo(f"\nFile: {file_path}\n")
+        for result in result_rows:
+            click.echo(
+                f"Detector: {result[0]}\n"
+                f"Impact: {result[1]}\n"
+                f"Details: {result[2]}\n"
+                f"Execution Paths (#Lines):\n{result[3]}\n"
+            )
+            impact_frequency[result[1]] = impact_frequency.get(result[1], 0) + 1
+    # print summary by impact label
+    click.echo("\nTotal issues:")
+    for impact, frequency in impact_frequency.items():
+        click.secho(f"{impact}: {frequency}", fg="yellow")
+
+
+def has_template_vars(path: Path) -> bool:
+    """
+    Check if the file contains template variables.
+
+    Args:
+        path (Path): The file path to check.
+
+    Returns:
+        bool: True if template variables are found, False otherwise.
+    """
+    content = path.read_text()
+    return bool(re.search(r"\bTMPL_\w*\b", content))
```

Comment:
Our beaker smart contract template generates TEAL with a comment like `// TMPL_DELETABLE` and `// TMPL_UPDATABLE`. Won't these be detected as incorrectly having template variables?